index/slashing/slashing.go

```diff
@@ -36,52 +35,59 @@ type SlashingIndex struct {
 	ds       datastore.TxnDatastore
 	signaler *signaler.Signaler
 
-	consensus ConsensusHistory
-	// ToDo: deal storage consensus, wait for Lotus implementation
+	lock  sync.Mutex
+	index Index
 
-	lock     sync.Mutex
 	ctx      context.Context
 	cancel   context.CancelFunc
 	finished chan struct{}
 	closed   bool
 }
 
-// ToDo: handle proper forking, recent history, etc?
-type ConsensusHistory struct {
-	// LastUpdated is the block height of the information
-	LastUpdated uint64
-	// History has slashed height History of storage miners
-	History map[string][]uint64
+type Index struct {
+	Tipset types.TipSetKey
+	Miners map[string]Info
+}
+
+type Info struct {
+	History []uint64
```

Comment:
`Height` was changed to `Tipset`.

The thing here is that `Height` isn't a correct description of the point we last updated the index, since the chain may fork. So, if we say `Last sync was at height XXX` we can't guarantee that is the same tipset of the current chain (if it was reorged by some heavier fork).

So... the correct description is the fingerprint of the tipset. This will become more apparent when #28  is done since is quite important. Is related with the talk with FC guys about `ChainNotify` receiving a tipset checkpoint.

This has some big implications... since if when the index resumes syncing it discovers that the last tipset isn't part of the current chain (because it was reorged), we should kind of do a full refresh. A better option is having some checkpoints of index state at lower heights to rebuild from there. I have some things in mind, but that's further work.